NAMING.md

```diff
@@ -0,0 +1,373 @@
+# Customizing schema, table, relationship, and service API naming
+
+## Guidelines
+
+- Model names should be singular and use `PascalCase`
+  - `as` should be used for all relationships with a `camelCase` names to be implicit and to override `Sequelize`'s default. This value is used for customizing the relationship name and will help determining the column name in the database.
+  - `foreignKey` can be used
+- Table names should be singular and use `snake_case`. Databases like Postgres requires wrapping all non-lower case table and field names with double quotes.
+- URLs should be singular or plural depends on relationships and use `kebab-case` for model names and includes
+- Request payloads and response bodies should be singular or plural depends on relationships and use:
+  - `PascalCase` for model names
+  - `camelCase` for relationships
+- Hatchify backend as well as [Sequelize](https://www.npmjs.com/package/sequelize) are using [Inflection](https://www.npmjs.com/package/inflection) for pluralization of names. It takes care of irregular nouns for us.
+
+## Examples
+
+### Base Example
+
+Given a pre-existing model
+
+```js
+const Account = {
+  name: "Account",
+  attributes: {
+    name: "STRING",
+  },
+  belongsTo: [{ target: "SalesPerson" }],
+}
+```
+
+A name like `SalesPerson` in the schema:
+
+```js
+const SalesPerson = {
+  name: "SalesPerson",
+  attributes: {
+    firstName: "STRING",
+  },
+}
+```
+
+The server will already have the following endpoints for accounts:
+
+1. `GET /api/accounts?fields[accounts]=name&filter[name]=Bitovi`
+1. `GET /api/accounts/:id`
+1. `PATCH /api/accounts/:id`
+1. `DELETE /api/accounts/:id`
+
+And the following endpoints for sales persons:
+
+1. `GET /api/sales-persons?fields[sales-persons]=name&filter[name]=John+Doe` (appending `s` regardless of the model name)
+1. `GET /api/sales-persons/:id`
+1. `PATCH /api/sales-persons/:id`
+1. `DELETE /api/sales-persons/:id`
+
+1. Creates a table `sales_person` with the columns `id` and `first_name` and a table `account` with the columns `id`, `name` and `sales_person_id`.
```

Comment:
In the markdown file this continued as part of the previous numbered list; I think you want them to be different.

Reply:
![image](https://github.com/bitovi/hatchify/assets/58667978/7634260e-d8af-4b8e-b232-133c82fc71df)